privacy-on-beam/pbeam/count.go

```diff
@@ -125,47 +120,30 @@ func Count(s beam.Scope, pcol PrivatePCollection, params CountParams) beam.PColl
 		newDecodePairInt64Fn(partitionT.Type()),
 		countPairs,
 		beam.TypeDefinition{Var: beam.XType, T: partitionT.Type()})
-
-	if params.specifiedPartitions.partitionsSpecified { // Partitions specified.
-		return addSpecifiedPartitionsForCount(s, epsilon, delta, maxPartitionsContributed,
-			params, noiseKind, countsKV)
-
+	// Add specified partitions, if partitions are specified.
+	if (params.partitionsCol).IsValid() { 
+		return addSpecifiedPartitionsForCount(s, epsilon, delta, maxPartitionsContributed, params, noiseKind, countsKV)
 	}
 	sums := beam.CombinePerKey(s,
-		newBoundedSumInt64Fn(epsilon, delta, maxPartitionsContributed, 0, params.MaxValue, noiseKind, false),
-		countsKV)
+		newBoundedSumInt64Fn(epsilon, delta, maxPartitionsContributed, 0, params.MaxValue, noiseKind, false), countsKV)
+	// Drop thresholded partitions.
 	counts := beam.ParDo(s, dropThresholdedPartitionsInt64Fn, sums)
-	return beam.ParDo(s, clampNegativePartitionsInt64Fn, counts)
-}
-
-func addSpecifiedPartitionsForCount(s beam.Scope, epsilon, delta float64,
-	maxPartitionsContributed int64, params CountParams, noiseKind noise.Kind, countsKV beam.PCollection) beam.PCollection {
-	partitionsCol := params.specifiedPartitions.partitionsCol
-	// Turn partitionsCol type PCollection<K> into PCollection<K, int64> by adding
-	// the value zero to each K.
-	prepareAddSpecifiedPartitions := beam.ParDo(s, addValuesToSpecifiedPartitionKeysInt64Fn, partitionsCol)
-	// Merge countsKV and prepareAddSpecifiedPartitions.
-	allPartitions := beam.Flatten(s, prepareAddSpecifiedPartitions, countsKV)
-	// Sum and add noise.
-	sums := beam.CombinePerKey(s,
-		newBoundedSumInt64Fn(epsilon, delta, maxPartitionsContributed, 0, params.MaxValue, noiseKind, true),
-		allPartitions)
-
-	finalPartitions := beam.ParDo(s, DereferenceValuesToInt64, sums)
 	// Clamp negative counts to zero and return.
-	return beam.ParDo(s, clampNegativePartitionsInt64Fn, finalPartitions)
-
+	return beam.ParDo(s, clampNegativePartitionsInt64Fn, counts)
 }
 
-func checkCountParams(params CountParams, epsilon, delta float64) error {
+func checkCountParams(params CountParams, epsilon, delta float64, noiseKind noise.Kind) error {
 	err := checks.CheckEpsilon("pbeam.Count", epsilon)
 	if err != nil {
 		return err
 	}
 	err = checks.CheckDeltaStrict("pbeam.Count", delta)
+	if (params.partitionsCol).IsValid() && noiseKind == noise.LaplaceNoise {
```

Comment:
You should make the `CheckDeltaStrict` check in the else statement for this if because otherwise if you have specified partitions and Laplace noise, you would return an error above.

(Also for other aggregations where applicable)

Reply:
I'm not sure that's the case. if err from 'checks.CheckDeltaStrict' is not nil and partitions are specified and noiseKind is specified,  then the value of err will change to 'checks.CheckNoDelta'. This all happens before we check if err is nil or not.

Reply:
Either way, I addressed this comment.